testsite/settings.py

```diff
@@ -199,3 +199,12 @@ def load_config(confpath):
 INTERNAL_IPS = ('127.0.0.1', '::1')
 
 NOCAPTCHA = True
+
+if not DEBUG:
+    DEFAULT_FILE_STORAGE = 'storages.backends.s3boto3.S3Boto3Storage'
```

Comment:
We don't actually need to do that in testsite (testsite should neither run in a production environment). What is important though is that the storage should chosen dynamically (multi-tenant). See from pages/utils.py:

    def get_default_storage_base(request, account=None, **kwargs):
        # default implementation
        storage_class = get_storage_class()
        try:
            _ = storage_class.bucket_name
            storage_kwargs = {}
            storage_kwargs.update(**kwargs)
            for key in ['access_key', 'secret_key', 'security_token']:
                if key in request.session:
                    storage_kwargs[key] = request.session[key]
            return storage_class(
                bucket=_get_bucket_name(account),
                location=_get_media_prefix(account),
                **storage_kwargs)
        except AttributeError:
            LOGGER.debug("``%s`` does not contain a ``bucket_name``"\
                " field, default to FileSystemStorage.", storage_class)
        return _get_file_system_storage(account)

Reply:
I'm not sure how this can be used in our situation. This function depends on the availability of `request` object. Finally, to leverage the `django-storages` infrastructure we need to provide a static value to `storage` argument of model field (there is no way to pass a function).

Reply:
What seems most appropriate is to use a `URLField` in the model and [code like the following on save](https://github.com/djaodjin/djaodjin-pages/blob/02e6340c773cc9bda9c10b08b20c438d190c90e9/pages/api/upload_media.py#L93).